src/hera/json_utils.py

```diff
@@ -0,0 +1,26 @@
+import json
+from typing import Union, Dict, List
+
+from pydantic import BaseModel
+
+
+JSONType = Union[None, BaseModel, bool, str, float, int, List['JSONType'], Dict[str, 'JSONType']]  # type: ignore
+
+
+def encode_json(value: JSONType) -> str:
+    if isinstance(value, BaseModel):
+        return value.json()
+    elif isinstance(value, str):
+        return value
+    else:
+        return json.dumps(value)
+
+
+def create_param_extraction(param_name: str) -> str:
+    return f"""
+try:
+    {param_name} = json.loads('''{{{{inputs.parameters.{param_name}}}}}''')
+except json.JSONDecodeError:
```

Comment:
Under what circumstances will `json.loads` throw a decode error here? Is the exception block dedicated to catching strings that are formatted as `'"s"'` vs '"'s'"', perhaps?

Reply:
Strings will always come through without any enclosing quotes, leading to the decode error. This is a result of [this line](https://github.com/argoproj-labs/hera-workflows/pull/100/files#diff-c5438bb6fa034bef3ea5a671e3609f3b0ce4caabe5656501d546ca2537d133bcR13-R14)